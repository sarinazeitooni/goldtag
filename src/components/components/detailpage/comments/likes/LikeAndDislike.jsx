import React, {useState} from 'react';
import '../style/like-and-dislike.scss';
import {useDispatch, useSelector} from "react-redux";
import ThumbDownOutlinedIcon from '@material-ui/icons/ThumbDownOutlined';
import ThumbUpAltOutlinedIcon from '@material-ui/icons/ThumbUpAltOutlined';
import actionTypes from "../../../../redux/actionTypes/actionTypes";
function LikeAndDislike({like,dislike}){
    localStorage.setItem('like' , like);
    localStorage.setItem('dislike' , dislike);
    const dispatch = useDispatch();
    let likes = useSelector((state) => state.likes.like);
    let dislikes = useSelector((state) => state.likes.dislike);
    function Dispatch(action){
        dispatch({
            type : action
        })
    }
    function addDislike(){
        Dispatch(actionTypes.Like);
    }
    function addLike(){
        Dispatch(actionTypes.Dislike);
    }
    return(
        <div className='like-dislike-container'>
            <button className='like-buttons' onClick={addDislike}>
                <ThumbUpAltOutlinedIcon/>
                <span className='counter-container'>{likes}</span>
            </button>
            <button className='like-buttons' onClick={addLike}>
                <ThumbDownOutlinedIcon/>
                <span className='counter-container'>{dislikes}</span>
            </button>
        </div>
    )
}
export default LikeAndDislike;