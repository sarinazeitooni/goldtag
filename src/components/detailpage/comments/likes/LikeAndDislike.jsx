import React from 'react';
import '../style/like-and-dislike.scss';
import {useDispatch, useSelector} from "react-redux";
import ThumbDownOutlinedIcon from '@material-ui/icons/ThumbDownOutlined';
import ThumbUpAltOutlinedIcon from '@material-ui/icons/ThumbUpAltOutlined';
import actionTypes from "../../../../redux/actionTypes/actionTypes";
function LikeAndDislike({id}){
    const dispatch = useDispatch();
    let likes = useSelector((state) => state.likes.like);
    let dislikes = useSelector((state) => state.likes.dislike);
    function Dispatch(action , e){
        dispatch({
            type : action,
            id : e
        })
    }
    function addDislike(e){
        Dispatch(actionTypes.Like , e);
        console.log(e,"dislike");
    }
    function addLike(e){
        Dispatch(actionTypes.Dislike , e);
        console.log(e,"like");
   }
    return(
        <div className='like-dislike-container'>
            <button className='like-buttons' onClick={()=>{addDislike(id)}}>
                <ThumbUpAltOutlinedIcon/>
                <span className='counter-container'>{likes}</span>
            </button>
            <button className='like-buttons' onClick={()=>{addLike(id)}}>
                <ThumbDownOutlinedIcon/>
                <span className='counter-container'>{dislikes}</span>
            </button>
        </div>
    )
}
export default LikeAndDislike;