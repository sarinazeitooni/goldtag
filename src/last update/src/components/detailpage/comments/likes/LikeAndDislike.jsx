import React, {useState} from 'react';
import '../style/like-and-dislike.scss';
import {useDispatch, useSelector} from "react-redux";
import ThumbDownOutlinedIcon from '@material-ui/icons/ThumbDownOutlined';
import ThumbUpAltOutlinedIcon from '@material-ui/icons/ThumbUpAltOutlined';
import actionTypes from "../../../../redux/actionTypes/actionTypes";
import ls from 'local-storage';
function LikeAndDislike({likeId,like,dislike}){
    const dispatch = useDispatch();
    const likeItem = useSelector((state) => state.likes.likes.id);
    let likeNum = useSelector((state) => state.likes.likes.id.like)
    let dislikeNum = useSelector((state) => state.likes.dislikes.id.dislike)
    const dislikeItem = useSelector((state) => state.likes.dislikes.id);
    // let likes = (likeItem == likeId) ? likeNum : 0;
    // let dislikes =(dislikeItem == likeId) ? dislikeNum : 0;
    function Dispatch(action){
        dispatch({
            type : action ,
            payload: {
                likeId
            }
        })
    }
    function addDislike(){
        Dispatch(actionTypes.Like);
    }
    function addLike(){
        Dispatch(actionTypes.Dislike);
    }
    return(
        <div className='like-dislike-container'>
            <button className='like-buttons' onClick={addDislike}>
                <ThumbUpAltOutlinedIcon/>
                <span className='counter-container'>{likeNum}</span>
            </button>
            <button className='like-buttons' onClick={addLike}>
                <ThumbDownOutlinedIcon/>
                <span className='counter-container'>{dislikeNum}</span>
            </button>
        </div>
    )
}
export default LikeAndDislike;