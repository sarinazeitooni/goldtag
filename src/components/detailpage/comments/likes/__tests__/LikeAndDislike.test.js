import React from 'react';
import {shallow} from 'enzyme';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import LikeAndDislike from "../LikeAndDislike";
describe("LikeAndDislike component", () => {
    const initialState = {}
    const mockStore = configureStore();
    let Store = mockStore(initialState);
    it("it should render", () => {
        const container = shallow(<Provider store={Store}><LikeAndDislike data={''} id={''}/></Provider>)
        expect(container).toMatchSnapshot();
    });
    it("it should render like-dislike-container div", () => {
        const container = shallow(<Provider store={Store}><LikeAndDislike data={''} id={''}/></Provider>)
        expect(container.find('div.like-dislike-container')).toBeTruthy();
    });
});