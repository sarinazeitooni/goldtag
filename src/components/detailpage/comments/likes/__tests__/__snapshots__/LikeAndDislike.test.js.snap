// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeAndDislike component it should render 1`] = `ShallowWrapper {}`;
